torch/_inductor/codegen/cpp_wrapper_cpu.py

```diff
@@ -1422,17 +1422,28 @@ def can_stack_allocate_buffer(self, buffer):
         )
 
     def make_buffer_free(self, buffer):
-        return (
-            ""
-            if isinstance(buffer.get_layout(), ir.MultiOutputLayout)
+        if (
+            isinstance(buffer.get_layout(), ir.MultiOutputLayout)
             or (V.graph.aot_mode and buffer.get_name() in self.stack_allocated_buffers)
             or (
                 config.use_minimal_arrayref_interface
                 and V.graph.aot_mode
                 and buffer.get_name() in V.graph.graph_inputs
             )
-            else f"{buffer.get_name()}.reset();"
-        )
+        ):
+            return ""
+        elif (
+            isinstance(buffer, ir.ExternKernel)
+            and isinstance(buffer.op_overload, torch._ops.OpOverload)
+            and isinstance(buffer.op_overload._schema.returns[0].type, torch.ListType)
```

Comment:
`returns[0]` works for your case, but maybe we should generalize the logic to all `returns`?

Reply:
I believe `len(returns) > 1` is the MultiOutputLayout case, and based on existing code we explicitly don't generate buffer free code for it.